satellite-imagery.md

```diff
@@ -0,0 +1,136 @@
+has certain tools and certain expectations about what they should be able to
+do with the data. As satellite instrument technology advances scientists are
+now having to learn how to handle more channels for each instrument and at
+spatial and temporal resolutions that were unheard of when they were learning
+how to use satellite data. If they are lucky scientists may have access to a
+high performance computing system, while the rest may have to settle for long
+execution times on their desktop or laptop machines.
+
+SatPy tries to make all of this processing easy to handle with simple
+interfaces and logical defaults to produce the best results. By optimizing the
+parts of the processing that typically take a lot of time and memory it is our
+hope that scientists can worry about the science and leave the annoying parts
+to SatPy.
+
+
+How Dask helps
+--------------
+
+SatPy was originally drawn to dask for its lazy evaluation and multi-threaded
+processing. Dask is used as the array backend for the xarray library's
+`DataArray` object which is used as the main data container object in SatPy.
+DataArray objects allow SatPy to keep the satellite imagery data, data
+coordinates, and other metadata all in one place. Before using xarray and dask
+SatPy used numpy masked arrays but as new satellite instruments were launched
+our users simply couldn't handle the amount of memory required to perform
+basic operations. The processing involved in SatPy requires a lot of
+intermediate variables/calculations and comparisons between different data
+arrays. With masked arrays and the right calculations you can quickly run out
+of memory on a 16-32GB machine with instruments launched in the last 5-10
+years.
+
+Dask allows SatPy to split arrays in to multiple chunks and process them in
+parallel, something that is extremely useful in satellite processing where
+pixels at a particular array index are all processed the same or compared with
+each other. By using `NaN` as a fill value in our dask arrays instead of having
+a separate mask array we reduced our memory consumption a ton. And due to the
+way dask schedules and caches tasks SatPy is able to reduce the amount of
+intermediate calculation results stored in memory at any given time. This
+makes it possible to do calculations on laptops that used to require high
+performance server machines.
+
+Lastly, developing software for satellite processing used to take a long time
+because failing at the end of the processing chain sometimes required
+reprocessing all of the previous steps, filling up memory, and then releasing
+it to start the next step of processing. With dask we can persist results in
+memory if we want or at the very least hold on to the dask graph for the steps
+already completed.
+
+
+Pain points when using dask
+---------------------------
+
+1. Dask is not numpy. Almost everything is supported or is close enough that
```

Comment:
Request changing "Dask" to "Dask array" in settings like this so as not to confuse the dask.dataframe / dask.delayed folks.